api/scpca_portal/models/original_file.py

```diff
@@ -216,6 +217,49 @@ def download_path(self) -> Path:
         """Return the remaining part of self.s3_key that's not the download_dir."""
         return self.s3_key_path.relative_to(self.download_dir)
 
+    @property
+    def local_file_path(self):
+        return settings.INPUT_DATA_PATH / self.s3_key_path
+
+    def _exchange_multiplexed_delimeter(self, output_file: str) -> Path:
+        """
+        Return the file path with the exchanged input and output multiplexed sample delimeters.
+        """
+        return Path(
+            output_file.replace(
+                common.MULTIPLEXED_SAMPLES_INPUT_DELIMETER,
+                common.MULTIPLEXED_SAMPLES_OUTPUT_DELIMETER,
+            )
+        )
+
+    def get_zip_file_path(self, download_config: Dict) -> Path:
+        """Return file path with requested directory structure according to download config."""
+        output_path = self._exchange_multiplexed_delimeter(self.s3_key)
+
+        # Project output paths are relative to project directory
+        output_path = output_path.relative_to(Path(self.project_id))
+
+        # Sample output paths are relative to sample directory
+        if download_config in common.SAMPLE_DOWNLOAD_CONFIGS.values():
+            sample_id_path = Path(
+                # Delimeter must be exchanged if file has multiplexed samples
+                self._exchange_multiplexed_delimeter(self.s3_key_info.sample_id_part)
+            )
+            return output_path.relative_to(sample_id_path)
+
+        # Transform merged and bulk project data files to no longer be nested in a merged directory
+        if self.is_merged:
+            return output_path.relative_to(common.MERGE_INPUT_DIR)
+        if self.is_bulk:
+            return output_path.relative_to(common.BULK_INPUT_DIR)
+
+        # Nest sample reports into individual_reports directory in merged download
+        # The merged summmary html file should not go into this directory
+        if download_config.get("includes_merged", False) and self.is_supplementary:
+            return Path("individual_reports") / output_path
+
+        return output_path
```

Comment:
Let's create a private version of _get_zip_file_path that does not handle the replacing of commas for multiplexed, and a public wrapper function that calls _get_zip_file_path and returns utils.path_replace(path, ',', '_')

Also, lets work off of s3_key_info.project_id_part when working with paths to be consistent with s3_key_info.sample_id_part

also, lets move individual_reports magic string to common like merge and bulk